lib/src/geometry/circle.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:math';
+import 'dart:ui';
+
+import '../../components.dart';
+import '../extensions/vector2.dart';
+import 'shape.dart';
+
+class Circle extends Shape {
+  /// The [definition] is how many percentages of the shortest edge of [size]
+  /// that the circle should cover.
+  double definition;
+
+  /// With this constructor you can create your [Circle] from a radius and
+  /// a position. Ir will also calculate the bounding rectangle [size] for the
+  /// [Circle].
+  Circle({
+    double radius,
+    Vector2 position,
+    double angle = 0,
+  }) : super(
+          position: position,
+          size: Vector2.all((radius ?? 0) * 2),
+          angle: angle,
+        );
+
+  /// This constructor is used by [HitboxCircle]
+  /// definition is the percentages of the shortest edge of [size] that the
+  /// circle should fill.
+  Circle.fromDefinition({
+    this.definition = 1.0,
+    Vector2 position,
+    Vector2 size,
+    double angle,
+  }) : super(position: position, size: size, angle: angle = 0);
+
+  double get radius {
+    return (min(size.x, size.y) / 2) * definition;
+  }
+
+  @override
+  void render(Canvas canvas, Paint paint) {
+    final localPosition = size / 2 + position;
+    final localRelativePosition = (size / 2)..multiply(relativePosition);
+    canvas.drawCircle(
+      (localPosition + localRelativePosition).toOffset(),
+      radius,
+      paint,
+    );
+  }
+
+  /// Checks whether the represented circle contains the [point].
+  @override
+  bool containsPoint(Vector2 point) {
+    return shapeCenter.distanceToSquared(point) < radius * radius;
+  }
+
+  /// Returns the locus of points in which the provided line segment intersect
+  /// the circle.
+  ///
+  /// This can be an empty list (if they don't intersect)
```

Comment:
Add commas between the list elements :)

Reply:
Didn't understand what you meant here?

Reply:
```
This can be an empty list (if they don't intersect), one point (if the line is tangent), or two points (if the line is secant).
```

Reply:
Ah, didn't realize there were more lines under the comment, you only see the ones above on the basic github view.